src/app/pages/incident-details-view/incident-details-view.component.ts

```diff
@@ -0,0 +1,532 @@
+  incident!: Incident;
+  incidentUpdates!: IncidentUpdateResponseData[];
+
+  // We create a copy of all the incident's values we might change so that
+  // we can properly restore them in case the user discards any changes.
+  incidentCopy: Incident = {};
+
+  // These sets only include some form of ID we can use to
+  // differentiate impacts/updates from each other. The actual
+  // pending objects are stored in impactsToAdd/updatesToAdd.
+  // Why? Because we can now quickly check if in the UI if
+  // an impact or update is pending by calling the Set.has
+  // method on the appropriate set. Furthermore, we can create
+  // an overall array for iteration in the UI by using the 
+  // spread operator.
+  pendingImpacts: Set<ComponentId> = new Set();
+  pendingUpdates: Set<number> = new Set();
+
+  // New entities to add to this incident. Accompanying ID data
+  // is stored in pendingImpacts and pendingUpdates, respectively.
+  impactsToAdd: Impact[] = [];
+  updatesToAdd: IncidentUpdateResponseData[] = [];
+
+  // Impacts and updates we wish to delete.
+  impactsToDelete: Set<ComponentId> = new Set();
+  updatesToDelete: Set<number> = new Set();
+
+  userAuthenticated: boolean = false;
+
+  edit: EditMode;
+
+  currentErrors: Map<ResultId, Result>;
+
+  startDate: string = "";
+  endDate: string = "";
+
+  // We track the highest order number so far to make sure we have a key
+  // we can use to differentiate IncidentUpdates from one another, even
+  // those that have not yet been send to the API server.
+  latestUpdateOrder: number = 0;
+
+  newIncident: boolean = false;
+  maintenanceEvent: boolean = false;
+
+  @ViewChild("editImpactDialog", {static: true})
+  private editImpactDialog!: ElementRef<EditImpactDialogComponent>;
+  @ViewChild("waitSpinnerDialog", {static: false})
+  private waitSpinnerDialog!: SpinnerDialogComponent;
+  @ViewChild("confirmDeleteDialog", {static: false})
+  private confirmDeleteDialog!: ConfirmDeleteDialogComponent;
+
+  _waitState: string = "";
+
+  ready: boolean = false;
+
+  constructor(
+    public data: DataService,
+    public util: UtilService,
+    public ip: IconProviderService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private security: OidcSecurityService,
+    private incidentService: IncidentService,
+    private toastr: ToastrService
+  ) {
+    this.edit = new EditMode();
+    this.edit.leaveEditFunction = this.dispatchEditLeave.bind(this);
+    this.currentErrors = new Map();
+  }
+
+  ngOnInit(): void {
+    // Check if the user is authenticated
+    this.security.checkAuth().subscribe(async response => {
+      this.userAuthenticated = response.isAuthenticated;
+      if (this.userAuthenticated) {
+        const token = await firstValueFrom(this.security.getAccessToken());
+        this.incidentService.configuration.withCredentials = true;
+        this.incidentService.defaultHeaders = this.incidentService.defaultHeaders.set("Authorization", `Bearer ${token}`);
```

Comment:
Maybe you want to set the default headers for the whole instance as soon as the user authenticates. 
This would authorize every request.

Reply:
I now do so in the login component. Will need to test this, though.